### PR TITLE
docs(react): update react functional component tutorial

### PR DESCRIPTION
**Pull request by FredZeng**
## Description
https://github.com/videojs/video.js/issues/7361

## Specific Changes proposed
Please list the specific changes involved in this pull request.

## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [x] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: docs/guides/react.md [hunk 1]
```diff
@@ -12,43 +12,55 @@ import "video.js/dist/video-js.css";
...
 
-  // This separate functional component fixes the removal of the videoelement 
-  // from the DOM when calling the dispose() method on a player
-  const VideoHtml = ( props ) => (
-    <div data-vjs-player>
-      <video ref={videoRef} className="video-js vjs-big-play-centered" />
-    </div>
-  );
+  React.useEffect(() => {
+    // make sure VjsPlayer only initial once
```
Comment (gkatsev) [inline docs/guides/react.md:19]: nit:
```suggestion
    // make sure Video.js player is only initialized once
```

File: docs/guides/react.md [hunk 2]
```diff
@@ -12,43 +12,55 @@ import "video.js/dist/video-js.css";
...
-      player = videojs( videoElement, options, () => {
+      const player = playerRef.current = videojs(videoElement, options, () => {
         console.log("player is ready");
+        onReady && onReady(player);
       });
+    } else {
+      // you can update player here [update player through props]
+      // const player = playerRef.current;
+      // player.autoplay(options.autoplay);
+      // player.src(options.sources);
```
Comment (gkatsev) [inline docs/guides/react.md:32]: 👍🏻

File: docs/guides/react.md [hunk 3]
```diff
@@ -12,43 +12,55 @@ import "video.js/dist/video-js.css";
...
     }
+  }, [options]);
+
+  // Dispose VjsPlayer when functional component actually unmount
```
Comment (gkatsev) [inline docs/guides/react.md:36]: nit:
```suggestion
  // Dispose the Video.js player when the functional component unmounts
```

File: docs/guides/react.md [hunk 4]
```diff
@@ -60,18 +72,40 @@ const App = () => {
       type: 'video/mp4'
     }]
   }
-  
+
+  const handlePlayerReady = (player) => {
+    playerRef.current = player;
+
+    // you can handle player's events here
```
Comment (gkatsev) [inline docs/guides/react.md:79]: nit:
```suggestion
    // you can handle player events here
```

File: docs/guides/react.md [hunk 5]
```diff
@@ -60,18 +72,40 @@ const App = () => {
...
+      console.log('player is waiting');
+    });
+
+    player.on('dispose', () => {
+      console.log('player will dispose');
+    });
+  };
+
+  // const changePlayerOptions = () => {
+  //   // you can update player through VjsPlayer instance
```
Comment (gkatsev) [inline docs/guides/react.md:90]: nit:
```suggestion
  //   // you can update the player through the Video.js player instance
```

**Review by gkatsev — COMMENTED** [review]
Thanks, @FredZeng, this is super helpful!
Just had a couple of nitpicks.

**Review by gkatsev — APPROVED** [review]
Thanks!

File: docs/guides/react.md [hunk 6]
```diff
@@ -60,18 +72,40 @@ const App = () => {
...
+  // };
+
   return (
     <>
       <div>Rest of app here</div>
-      
-      <VideoJS options={videoJsOptions}/>
-      
+
+      <VideoJS options={videoJsOptions} onReady={handlePlayerReady} />
```
Comment (brandonocasey) [inline docs/guides/react.md:103]: 👍🏻

**Review by brandonocasey — APPROVED** [review]

**Comment by gkatsev** [comment]
Thanks!

**Pull request merged**